server/src/test/java/org/elasticsearch/index/mapper/BooleanFieldMapperTests.java

```diff
@@ -257,15 +259,24 @@ public void testDimensionIndexedAndDocvalues() {
         }
     }
 
-    public void testDimensionMultiValuedField() throws IOException {
+    public void testDimensionMultiValuedField() throws Throwable {
         XContentBuilder mapping = fieldMapping(b -> {
             minimalMapping(b);
             b.field("time_series_dimension", true);
         });
-        DocumentMapper mapper = randomBoolean() ? createDocumentMapper(mapping) : createTimeSeriesModeDocumentMapper(mapping);
+        IndexMode indexMode = randomFrom(IndexMode.values());
```

Comment:
Maybe not randomize `indexMode` here and in other similar tests? I would prefer to have one unit test testing the case where the validation kicks in and another whether the validation doesn't kick in (in that test randomizing between standard and logsdb does make sense).

This should avoid the situation where someone thinks tests passes and then only after re-running would figure out that a change doesn't make sense.

Reply:
Makes sense, I've adjusted the tests.